styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -161,6 +162,7 @@ public WorkflowResource(Storage storage, WorkflowValidator workflowValidator,
 
     final Workflow workflow = Workflow.create(componentId, workflowConfig);
 
+    workflowActionAuthorizer.authorizeCreateOrUpdateWorkflowAction(workflow);
```

Comment:
I am not sure about the double actions as we invoke the `workflowActionAuthorizer` object twice with `authorizeCreateOrUpdateWorkflowAction` and `authorizeWorkflowAction`. Would it be simpler to just call `authorizeCreateOrUpdateWorkflowAction` with inputs `ac` and `workflow` and have the logic for normal `serviceAccountUsageAuthorizer` and `actionAuthorizer`?

Reply:
@tnsetting  I actually though about it. From a first glance it make sense to refactor the call directly into the `workflowActionAuthorizer` but the `authorizeWorkflowAction` method as it is written today is identical for all 3 calls (Create/update, Patch and Delete). Thus, one need to either use a specific function (like I did) or pass a hardcoded argument into the `authorizeWorkflowAction` function to indicate which extra action to take for the call. In the sake of clarity I would instead rename the `authorizeWorkflowAction` method to something more explicit like `authorizeUserForWorkflow`.
That way we have clear distinction between authorizing the user itself / service account on one side and the action taken on the other.